src/saveToCase.js

```diff
@@ -0,0 +1,475 @@
+        }
+    }
+
+    var propertyWidget = function (mug, options) {
+            var widget = widgets.normal(mug, options),
+                id = options.id;
+
+            widget.kvInput = $('<div class="control-row" />').attr('name', id);
+
+            widget.getControl = function () {
+                return widget.kvInput;
+            };
+
+            widget.refreshControl = function () {
+                widget.setValue(widget.getValue());
+            };
+
+            return widget;
+        },
+        saveCasePropWidget = function (mug, options) {
+            var widget = propertyWidget(mug, options);
+
+            widget.setValue = function (value) {
+                value = _.isUndefined(value) ? {} : value;
+                widget.kvInput.html(widget_update_case({
+                    props: value
+                }));
+                widget.kvInput.find('input').bind('change keyup', function () {
+                    widget.handleChange();
+                });
+                widget.kvInput.find('.fd-add-update-property').click(function (e) {
+                    widget.refreshControl();
+                    e.preventDefault();
+                });
+                widget.kvInput.find('.fd-remove-update-property').click(function (e) {
+                    $(this).parent().parent().parent().remove();
+                    widget.refreshControl();
+                    widget.save();
+                    e.preventDefault();
+                });
+            };
+
+            widget.getValue = function () {
+                var currentValues = {};
+                _.each(widget.kvInput.find('.fd-update-property'), function (kvPair) {
+                    var $pair = $(kvPair);
+                    currentValues[$pair.find('.fd-update-property-name').val()] = {
+                        calculate: $pair.find('.fd-update-property-source').val(),
+                        relevant: $pair.find('.fd-update-property-relevant').val(),
+                    };
+                });
+                return currentValues;
+            };
+
+            return widget;
+        };
+
+    var CASE_XMLNS = "http://commcarehq.org/case/transaction/v2",
+        saveToCaseMugOptions = {
+            typeName: 'Save to Case',
+            isTypeChangeable: false,
+            isDataOnly: true,
+            supportsDataNodeRole: true,
+            icon: 'icon-save',
+            init: function (mug, form) {},
+            spec: {
+                xmlnsAttr: { presence: "optional" },
+                "date_modified": {
+                    lstring: "Date modified",
+                    visibility: 'visible',
+                    presence: 'required',
+                    widget: widgets.xPath,
+                    validationFunc: function (mug) {
+                        if (mug.p.date_modified === "") {
+                            return "Date Modified is required";
```

Comment:
I think this is not necessary (unless you don't like the default error message). Validation for `presence="required"` is done for all mugs in [mugs.js](https://github.com/dimagi/Vellum/blob/save-to-case-on-master/src/mugs.js#L23-L25).

Aside: if the default error message is the problem, I think it would be reasonable to change it globally to use a format like `(spec.lstring || ruleKey) + " is required"`. It's already been changed on the [mug messages PR](https://github.com/dimagi/Vellum/blob/3b17849889117e21264739ee95779fe8af6c29af/src/mugs.js#L121).